src/Microsoft.TemplateEngine.Edge/GlobalSettingsTemplatesSources/GlobalSettingsTemplatesSourcesProvider.cs

```diff
@@ -140,69 +140,71 @@ public async Task<IReadOnlyList<UpdateResult>> UpdateAsync(IEnumerable<UpdateReq
                 IEnumerable<UpdateRequest> updatesToApply = updateRequests.Where(request => request.Version != request.Source.Version);
 
                 return await Task.WhenAll(updatesToApply.Select(
-                    async updateRequest => UpdateResult.FromInstallResult(updateRequest, await InstallAsync(updateRequest.Source, updateRequest.Version).ConfigureAwait(false)))).ConfigureAwait(false);
+                    async updateRequest => await UpdateAsync(updateRequest).ConfigureAwait(false))).ConfigureAwait(false);
```

Comment:
`async updateRequest => await UpdateAsync(updateRequest).ConfigureAwait(false))).ConfigureAwait(false);`
could be
`updateRequest => UpdateAsync(updateRequest))).ConfigureAwait(false);`
So state machine for async method is not needed.